packages/linter/tests/lib.ts

```diff
@@ -0,0 +1,224 @@
+const FIXTURES = "fixtures";
+
+import { existsSync, readFile, readFileSync } from "fs";
+
+import * as cheerio from "cheerio";
+import debug from "debug";
+import { diffJson as diff } from "diff";
+import { back as nockBack } from "nock";
+import { default as fetch } from "node-fetch";
+
+import throat from "throat";
+import { Status, Result, LintMode, guessMode } from "../src";
+import { RuleConstructor } from "../src/rule";
+
+const log = debug("linter");
+
+nockBack.fixtures = `${__dirname}/${FIXTURES}`;
+
+export let COUNT = 1;
+
+// Need to throttle to one run at a time because nock() works by monkey patching
+// the (global) http.* object, which means it can't run in parallel.
```

Comment:
I've switched from using nock to `fetch-mock` + dependency injection for amp-toolbox. This solves many annoying issues with nock.

Reply:
That might be a good idea… Does `fetch-mock` also auto-create/generate mocks if they're missing? That's a useful feature of nock, in the cases where multiple resources are downloaded.